File: 1248/py/wrnlb.py
from typing import (
    List,
    Dict,
)

class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        cur_sum: int = 0
        res: int = 0
        sum: Dict[int, int] = {cur_sum: 1}

        for num in nums:
            cur_sum += num % 2
            res += sum.get(cur_sum - k, 0)
            sum[cur_sum] = sum.get(cur_sum, 0) + 1

        return res




def main() -> None:
    nums: List[int] = [1,1,2,1,1]
    k: int = 3

    res: int = Solution().numberOfSubarrays(nums, k)
    print(res)


if __name__ == '__main__':
    main()

File: 1894/ts/wrnlb.ts
function chalkReplacer(chalk: number[], k: number): number {
    let sum: number = 0
    for (let i = 0; i < chalk.length; i++) {
        sum += chalk[i]
        if (sum > k) {
            return i
        }
    }
    k %= sum
    sum = 0
    for (let i = 0; i < chalk.length; i++) {
        sum += chalk[i]
        if (sum > k) {
            return i
        }
    }
    return 0
}


(() => {
    const chalk: number[] = [3,4,1,2]
    const k: number = 25
    const res: number = chalkReplacer(chalk, k)
    console.log(res)
})()
